datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -2116,6 +2171,48 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn join_right_semi() -> Result<()> {
+        let session_ctx = SessionContext::new();
+        let task_ctx = session_ctx.task_ctx();
+        let left = build_table(
+            ("a2", &vec![10, 20, 30, 40]),
+            ("b1", &vec![4, 5, 6, 5]), // 5 is double on the left
```

Comment:
recommend calling this `b2` not `b1` for clarity

Reply:
makes sense